app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt

```diff
@@ -64,7 +64,13 @@ class ClipboardHandler(val context: Context) {
     internal fun containsURL(): Boolean {
         return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
             val description = clipboard.primaryClipDescription
-            val score = description?.getConfidenceScore(TextClassifier.TYPE_URL) ?: 0F
+            // An IllegalStateException is thrown if the url is too long.
+            val score =
+                try {
+                    description?.getConfidenceScore(TextClassifier.TYPE_URL) ?: 0F
+                } catch (e: IllegalStateException) {
+                    0F
+                }
             score >= 0.7F
```

Comment:
@csadilek  I added the `try ` as I noticed that `getConfidenceScore` was throwing an `IllegalStateException` when the URL was too long.

Reply:
> 👍 I think this is a good immediate solution without losing any functionality, and we can improve later. I noticed two things though
> 
>     * After filling in from clipboard the cursor is at the beginning instead of at the end. After pasting text the user expectation is for the cursor to be at the end of the text.
> 
>     * The keyboard pops up and down when filling in.

Added [this](https://github.com/mozilla-mobile/fenix/pull/22391/commits/27c153937b93fa0a059ff73b26adcc5b275925cc#) separated commit to address the listed issues

Reply:
Tested again, works great now!